File: backend/app/schemas/meeting.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class Attendee(BaseModel):
    id: str = ""
    name: str
    email: str
    type: Literal["required", "optional"] = "required"
    status: Literal["accepted", "tentative", "declined", "no_response"] = "no_response"


class Meeting(BaseModel):
    id: str
    title: str
    description: str | None = None
    organiser: str
    attendees: list[Attendee] = Field(default_factory=list)
    start_time: datetime
    end_time: datetime
    location: str | None = None
    category_color: str = "BLUE"
    is_manda_created: bool = False
    created_at: datetime
    updated_at: datetime


class MeetingListResponse(BaseModel):
    meetings: list[Meeting]
    total: int


class MeetingCreateRequest(BaseModel):
    title: str
    description: str | None = None
    organiser: str
    attendees: list[Attendee] = Field(default_factory=list)
    start_time: datetime
    end_time: datetime
    location: str | None = None
    category_color: str = "BLUE"
    is_manda_created: bool = False


class GenerateDescriptionRequest(BaseModel):
    prompt: str
    generate_purpose: bool = True
    generate_outcomes: bool = True
    generate_agenda: bool = True
    personality: Literal["formal", "casual", "corporate", "friendly", "direct"] = "formal"
    context_prompt: str = ""
    template: str | None = None


class RewriteDescriptionRequest(BaseModel):
    current_description: str
    prompt: str
    personality: Literal["formal", "casual", "corporate", "friendly", "direct"] = "formal"
    context_prompt: str = ""


class MandaMeetingSettings(BaseModel):
    id: str = ""
    auto_color: str = "BLUE"
    personality: Literal["formal", "casual", "corporate", "friendly", "direct"] = "formal"
    context_prompt: str = ""
    enabled_templates: list[str] = Field(
        default_factory=lambda: [
            "tmpl-1on1",
            "tmpl-standup",
            "tmpl-kickoff",
            "tmpl-quarterly",
            "tmpl-brainstorm",
            "tmpl-retro",
        ]
    )
    company_branding_enabled: bool = False
    brand_color: str | None = None
    brand_header_text: str | None = None
    brand_footer_text: str | None = None
    safety_checks: list[str] = Field(
        default_factory=lambda: ["missing_agenda", "missing_purpose_outcome"]
    )


class MandaMeetingSettingsUpdate(BaseModel):
    auto_color: str | None = None
    personality: Literal["formal", "casual", "corporate", "friendly", "direct"] | None = None
    context_prompt: str | None = None
    enabled_templates: list[str] | None = None
    company_branding_enabled: bool | None = None
    brand_color: str | None = None
    brand_header_text: str | None = None
    brand_footer_text: str | None = None
    safety_checks: list[str] | None = None


class MeetingTemplate(BaseModel):
    id: str
    name: str
    content: str
    is_builtin: bool = False
    user_id: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class MeetingTemplateCreateRequest(BaseModel):
    name: str
    content: str

File: frontend/src/types/meeting.ts
export interface Attendee {
  id: string;
  name: string;
  email: string;
  type: "required" | "optional";
  status: "accepted" | "tentative" | "declined" | "no_response";
}

export interface Meeting {
  id: string;
  title: string;
  description: string | null;
  organiser: string;
  attendees: Attendee[];
  start_time: string;
  end_time: string;
  location: string | null;
  category_color: string;
  is_manda_created: boolean;
  created_at: string;
  updated_at: string;
}

export interface MeetingListResponse {
  meetings: Meeting[];
  total: number;
}

export interface MeetingCreateRequest {
  title: string;
  description?: string | null;
  organiser: string;
  attendees: Attendee[];
  start_time: string;
  end_time: string;
  location?: string | null;
  category_color: string;
  is_manda_created: boolean;
}

export interface MandaMeetingSettings {
  id: string;
  auto_color: string;
  personality: "formal" | "casual" | "corporate" | "friendly" | "direct";
  context_prompt: string;
  enabled_templates: string[];
  company_branding_enabled: boolean;
  brand_color: string | null;
  brand_header_text: string | null;
  brand_footer_text: string | null;
  safety_checks: string[];
}

export interface MandaMeetingSettingsUpdate {
  auto_color?: string;
  personality?: "formal" | "casual" | "corporate" | "friendly" | "direct";
  context_prompt?: string;
  enabled_templates?: string[];
  company_branding_enabled?: boolean;
  brand_color?: string | null;
  brand_header_text?: string | null;
  brand_footer_text?: string | null;
  safety_checks?: string[];
}

export interface MeetingTemplate {
  id: string;
  name: string;
  content: string;
  is_builtin: boolean;
  user_id: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface GenerateDescriptionRequest {
  prompt: string;
  generate_purpose: boolean;
  generate_outcomes: boolean;
  generate_agenda: boolean;
  personality: "formal" | "casual" | "corporate" | "friendly" | "direct";
  context_prompt: string;
  template?: string | null;
}

export interface RewriteDescriptionRequest {
  current_description: string;
  prompt: string;
  personality: "formal" | "casual" | "corporate" | "friendly" | "direct";
  context_prompt: string;
}
